_posts/2018/2018-06-18-bridging-the-gap-between-literate-programming-and-teaching-modules.markdown

```diff
@@ -0,0 +1,94 @@
+---
+layout: "post"
+title: "Bridging the gap between literate programming and teaching modules"
+date: "2018-06-18 11:07"
+author: Tania Allard
+tags:
+  - use-case
+  - best-practice
+  - open-science
+  - teaching
+  - jupyter
+---
+
+## Background
+The Jupyter notebook is a popular tool used by thousands of individuals and organisations all over the world. The notebook popularity is associated
+with the paradigm of **literate programming**, first introduced by
+**Donald E. Knuth**. Under such a paradigm the (computer) program is treated like
+a piece of literature addressed to humans rather than to computers.
+As a consequence, such programs focus on the logic and flow of human thought.
+At the same time, the programs integrate both the data analysis and textual documentation,
+making the programs more robust, more portable, and easier to maintain than
+programs are solely written in high-level languages.
```

Comment:
'programs that **are** solely....'

Reply:
with the paradigm of **literate computing**; this is a variant of the paradigm of **literate programming**  introduced in by Donald Knuth; under the latter .... Similarly, a literate computing document integrates both ...